tests/test_octodns_source_axfr.py

```diff
@@ -53,6 +54,19 @@ def test_zonefiles_with_extension(self):
         source.populate(valid)
         self.assertEquals(1, len(valid.records))
 
+    def test_zonefiles_without_extension(self):
+        try:
+            copyfile('./tests/zones/unit.tests.tst',
+                     './tests/zones/unit.tests.')
+        except:
```

Comment:
@ttk I'd appreciate it if you could provide the exact exception that's thrown by copyfile when it tries to copy to a filename with a `.` on the end in windows so that we don't have to catch all exceptions here. 

I would consider getting a better/tighter exception here a prerequisite for shipping this so that we can catch other/unrelated problems and fail the test rather than skip.

Reply:
Unfortunately the `copyfile()` doesn't raise any exception, instead it creates the following file: `./tests/zones/unit.tests` (transparently removes the trailing dot).

I had to do something like this to raise an exception:
```
--- a/tests/test_octodns_source_axfr.py
+++ b/tests/test_octodns_source_axfr.py
@@ -17,6 +17,7 @@ from octodns.source.axfr import AxfrSource, AxfrSourceZoneTransferFailed, \
     ZoneFileSource, ZoneFileSourceLoadFailure
 from octodns.zone import Zone
 from octodns.record import ValidationError
+import os


 class TestAxfrSource(TestCase):
@@ -58,6 +59,10 @@ class TestZoneFileSource(TestCase):
         try:
             copyfile('./tests/zones/unit.tests.tst',
                      './tests/zones/unit.tests.')
+
+            if os.path.exists('./tests/zones/unit.tests'):
+                raise Exception("Files ending in a . not supported")
+
         except:
             self.skipTest('Unable to create unit.tests. (ending with .) so '
                           'skipping default filename testing.')
```
This isn't great.  I first tried `if not os.path.exists('./tests/zones/unit.tests.'):` but that returned `True`.

Reply:
> This isn't great. I first tried `if not os.path.exists('./tests/zones/unit.tests.'):` but that returned `True`.

Yuk.

I'll probably rework this a bit and get rid of the try/catch then since it's not going to except and just skip test if the file without the `.` exists.